liby2/src/include/y2/Y2Namespace.h

```diff
@@ -132,11 +129,69 @@ class Y2Namespace {
 
     // pop all local variables from stack, uses SymbolEntry::pop()
     void popFromStack ();
-    
+
     // ensure that the namespace is initialized
     virtual void initialize ();
 
 };
 
+#define ERROR_NAMESPACE_NAME "_ErrorNamespace"
+class Y2ErrorNamespace : public Y2Namespace
+{
+public:
+
+  /**
+   * constructs new namespace in case when error happen
+   *
+   * @param summary short summary what error happen
+   * @param details can be longer details for error like e.g. backtrace
+   */
+  Y2ErrorNamespace(const std::string & summary, const std::string & details);
+
+  ~Y2ErrorNamespace();
+
+  /**
+   * shot summary what error happen
+   */
+  const std::string & summary() const;
+
+  /**
+   * longer details for error
+   */
+  const std::string & details() const;
+
+  const string name () const { return ERROR_NAMESPACE_NAME; }
+
+  const string filename () const { return "virtual_error_place"; }
+
+  /**
+   * creating funcition call for error namespace always fail
+   */
+  virtual Y2Function* createFunctionCall (const string name, constFunctionTypePtr type)
+  {
+    return NULL;
+  }
+
+  /**
+   * Does nothing for error namespace
+   */
+  YCPValue evaluate (bool cse = false);
+
+private:
+
+  string summary_;
+
+  string details_;
+};
+
+inline bool isErrorNamespace(const Y2Namespace * inst)
```

Comment:
well, I use helpers as it is more flexible if we change in future implementation. If we say, we support only this helpers, we have much more flexibility and much more easier extension. but if you prefer dynamic cast, it is fine for me